can/interfaces/usb_can_analyzer/usb_can_analyzer.py

```diff
@@ -0,0 +1,276 @@
+               200000: 0x06,
+               125000: 0x07,
+               100000: 0x08,
+               50000: 0x09,
+               20000: 0x0A,
+               10000: 0x0B,
+               5000: 0x0C
+              }
+    
+    FRAMETYPE = {
+                 "STD":0x01,
+                 "EXT":0x02
+                }
+
+    OPERATIONMODE = {
+                     "normal":0x00,
+                     "loopback":0x01,
+                     "silent":0x02,
+                     "loopback_and_silent":0x03
+                    }
+
+    def __init__(self, channel, baudrate=2000000, timeout=0.1, rtscts=False,
+                 frame_type='STD', operation_mode='normal', bit_rate=500000,
+                 *args, **kwargs):
+        """
+        :param str channel:
+            The serial device to open. For example "/dev/ttyS1" or
+            "/dev/ttyUSB0" on Linux or "COM1" on Windows systems.
+
+        :param int baudrate:
+            Baud rate of the serial device in bit/s (default 115200).
+
+            .. warning::
+                Some serial port implementations don't care about the baudrate.
+
+        :param float timeout:
+            Timeout for the serial device in seconds (default 0.1).
+
+        :param bool rtscts:
+            turn hardware handshake (RTS/CTS) on and off
+
+        """
+        self.bit_rate = bit_rate
+        self.frame_type = frame_type
+        self.op_mode = operation_mode
+        self.filter_id = bytearray([0x00, 0x00, 0x00, 0x00])
+        self.mask_id = bytearray([0x00, 0x00, 0x00, 0x00])
+        if not channel:
+            raise ValueError("Must specify a serial port.")
+
+        self.channel_info = "Serial interface: " + channel
+        self.ser = serial.Serial(
+            channel, baudrate=baudrate, timeout=timeout, rtscts=rtscts)
+
+        super(CanAnalyzer, self).__init__(channel=channel, *args, **kwargs)
+        self.init_frame
+
+    def shutdown(self):
+        """
+        Close the serial interface.
+        """
+        self.ser.close()
+
+    def init_frame(self, timeout=None):
+
+        byte_msg = bytearray()
+        byte_msg.append(0xAA)     # Frame Start Byte 1
+        byte_msg.append(0x55)     # Frame Start Byte 2
+
+        byte_msg.append(0x12)     # Initialization Message ID
+
+        byte_msg.append(CanAnalyzer.BITRATE[self.bit_rate])  # CAN Baud Rate
+        byte_msg.append(CanAnalyzer.FRAMETYPE[self.frame_type]) 
+
+        for i in range(0, 4):
+            byte_msg.append(self.filter_id[i])
+
+        for i in range(0, 4):
+            byte_msg.append(self.mask_id[i])
+
+        byte_msg.append(CanAnalyzer.OPERATIONMODE[self.op_mode])
+        
+        byte_msg.append(0x01)
+
+        for i in range(0, 4):
+            byte_msg.append(0x00)
+
+        print "byte_str = " + str(byte_msg[2:])
+
+        crc = Crc8Darc.calc(byte_msg[2:])
+        crc_byte = struct.pack('B', crc)
+
+        print "crc_byte: " + str(crc_byte) + " length: " + str(len(crc_byte))
```

Comment:
Replace this with a `logging` call instead. Same for other places.

Reply:
done.

Reply:
I've added debug level logs of written data as binary, let me know if would be better to format it?  I hesitated to add another import.  maybe binascii.hexlify()?

Reply:
I think it's fine.